pages/developers/blueprint-event-listeners/custom-listeners.mdx

```diff
@@ -2,6 +2,78 @@
 title: Custom Event Listeners
 ---
 
-# Custom Event Listeners
+To create an event listener, begin by defining a struct or enum which will listen to events:
 
-This section describes how to implement custom event listeners.
+```rust
+use gadget_sdk::event_listener::EventListener;
+use async_trait::async_trait;
+
+/// Define a simple event listener that ticks every MS milliseconds.
+pub struct Ticker<const MSEC: usize> {
+    additional_delay: u64,
+}
+```
+
+Then, define a context for the event listener:
+
+```rust
+#[derive(Copy, Clone)]
+pub struct MyContext {
+    pub additional_delay: u64,
+}
+```
+
+Next, implement `EventListener` for `Ticker`:
+
+```rust
+/// Implement the [`EventListener`] trait for the Ticker struct. [`EventListener`] has two type parameters:
+/// 
+/// - T: is the type of the event that the listener listens for. This can be anything that is Send + Sync + 'static.
```

Comment:
I don't see `T` listed anywhere. Docs need improvements. There's a lot of comments here that maybe are better outside of the code block and instead in markdown. I prefer the code implementation to be clean and code focused rather than comment focused. Since these are dev docs, those comments can be outside of the code blocks.